_appmap/test/helpers.py

```diff
@@ -15,3 +15,14 @@ class DictIncluding(dict):
 
     def __eq__(self, other):
         return other.items() >= self.items()
+
+
+class HeadersIncluding(dict):
+    """Like DictIncluding, but key comparison is case-insensitive."""
+
+    def __eq__(self, other):
+        for k in self.keys():
+            v = other.get(k, other.get(k.lower(), None))
+            if v is None:
+                return False
+        return True
```

Comment:
I don't really like the idea of changing the data that's getting emitted by the app/framework. I believe the Capitalized-Hyphenated format is conventional -- the [standard](https://datatracker.ietf.org/doc/html/rfc9110#name-field-names) says they're case-insensitive, right?

Reply:
Capitalized-Hyphenated is conventional; whatever the framework is using is likely its implementation detail. Why the reluctance to canonicalize it?

Reply:
I just don't see any benefit to making the change. It's not going to improve the user's (or Navie's) understanding of the code. And, it might lead to confusion, if the user headers in the AppMap that don't match what a `print` in the code would show.